pkg/internalapis/cnsoperator/triggercsifullsync/v1alpha1/triggercsifullsync_types.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2021 The Kubernetes authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TriggerCsiFullSyncSpec is the spec for TriggerCsiFullSync
+type TriggerCsiFullSyncSpec struct {
+	// TriggerSync gives an option to trigger full sync on demand.
+	// Inorder to trigger a full sync, user can set any random value.
+	TriggerSync int `json:"triggerSync"`
+}
+
+// TriggerCsiFullSyncStatus contains the status for a TriggerCsiFullSync
+type TriggerCsiFullSyncStatus struct {
+	// InProgress indicates whether a CSI full sync is in progress.
+	// If full sync is completed this field will be unset.
+	InProgress bool `json:"inProgress,omitempty"`
+
+	// LastSuccessfulStartTimeStamp indicates last successful full sync start timestamp.
+	LastSuccessfulStartTimeStamp time.Time `json:"lastSuccessfulStartTimeStamp,omitempty"`
+
+	// LastSuccessfulEndTimeStamp indicates last successful full sync end timestamp.
+	LastSuccessfulEndTimeStamp time.Time `json:"lastSuccessfulEndTimeStamp,omitempty"`
+
+	// LastRunStartTimeStamp indicates last run full sync start timestamp.
+	// This timestamp can be either the successful or failed full sync start timestamp.
+	LastRunStartTimeStamp time.Time `json:"lastRunStartTimeStamp,omitempty"`
+
+	// LastRunEndTimeStamp indicates last run full sync end timestamp.
+	// This timestamp can be either the successful or failed full sync end timestamp.
+	LastRunEndTimeStamp time.Time `json:"lastRunEndTimeStamp,omitempty"`
```

Comment:
In addition to the status of the last full sync, it will also be useful to keep a record of previous full sync status.

Reply:
If `LastSuccessfulStartTimeStamp` and `LastRunEndTimeStamp` are not same, then it means that last full sync has failed. If they are same, then it means that last one is successful.

Given this, I don't think we would need any additional param to record of previous full sync status.